src/arrayMethodJoin.js

```diff
@@ -4,8 +4,34 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    const addition = String(separator);
+    let newStr = '';
+
+    if (this.length === 0) {
+      return newStr;
+    }
+
+    for (let i = 0; i < this.length - 1; i++) {
+      switch (this[i]) {
+        case null:
+          newStr += '' + addition;
+          break;
+        case undefined:
+          newStr += '' + addition;
+          break;
+        default:
+          newStr += this[i] + addition;
+          break;
+      }
```

Comment:
switch case is redundant here
just add one check for the current element, and add element to the result string if it's not null of undefined